resource-managers/kubernetes/docker/src/main/dockerfiles/spark/sources.list

```diff
@@ -0,0 +1,20 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# This file is required for switching to https url based mirrors, See SPARK-31200, for more info.
+deb https://deb.debian.org/debian buster main
```

Comment:
https is good of course, but dumb question, was this the repo already being used, the Debian one?

Reply:
Not a dumb question, I should have mentioned it earlier. Yes, this was the one already being used. It is also suggested by https://www.debian.org/mirror/list , when the location is not known.

Reply:
I guess I mean more like: we don't currently have any reference to Debian or Buster in Spark; where is that coming from? just a property of the base image being used here?

Reply:
ahh! gotcha. Tomorrow oracle could be using some other base image. In that case, it might not even be of debian type. 
Then the fix would be a bit more than this, first check if it is a debian image and is using mirrors with http URLs, and then replace it with the `https` ones.